### ISSUE TITLE
Creating distribution of the developer version; crash because resource not found

### ISSUE THREAD

**Issue by Avalancs** [MovingBlocks/DestinationSol#181]
### What you were trying to do
I wanted to package the latest development version with the 'dist' and 'distZip' Gradle targets to be able to distribute it

### What actually happened
When the game is started it will crash with the error message missing src/main/resources/icon.png.

### How to reproduce
- run the distZip gradle task
- open the <project directory>\desktop\build\distributions\app folder
- start sol.exe

My discoveries:
- I see that instead of the old version's res folder now the assets are packaged inside the jar file.
- The path src/main/resources/icon.png can mean the program is actually looking for the file instead of using the classloader?
- icon.png, debugOptions.ini, log4j.properties, module.info, prevShip.ini and settings.ini seems to be added twice to <project directory>\desktop\build\distributions\app\libs\sol.jar. I had a similar problem once where gradle's distribution task had two rules which packaged the same files. I will investigate further tomorrow.

### Log details and game version
commit 9efb5ac0cc985546ac2fc211c64dd600b4ca92bd

**Comment by Cervator** [comment]
Thanks for the details! Yes this appeared after the GSOC project and we haven't had a chance yet to figure out what isn't going to the right place. Probably isn't too tricky a fix so if you can figure it out that'd be excellent :-)

**Comment by Avalancs** [comment]
Found the reason for the duplicated files in the jar. It's engine/build.gradle
```
sourceSets {
    // Main is almost everything - the true engine
    main {
        java {
            output.classesDir 'build/classes'
        }
        // This makes the resources (assets, 3d wizardry includes, etc) also go to build/classes
        output.resourcesDir 'build/classes'
}
```

Since the project has the 'java' plugin enabled it automatically has build/classes as source set hence this declaration makes it compress the files twice. Removing it fixes the duplication issue, but does not fix the original problem when starting the game, so I will continue with the investigation

**Comment by Avalancs** [comment]
Okay, so the problem is that when finding resources there are a lot of if-elses that add engine/src/main/resources to the beginning which work well in the development environment (since the files are physically there) but after packaging the files are not there but in the jar.

For engine resources which are included in the jar, the gfx FileHandles had to be set to FileType=classpath, and the *FileFormat classes had to be written that if they don't find a resource as a physical file then they should look inside the classpath. The problem is that when trying to load an asset the AbstractAssetFileFormat class does not know if the asset comes from a core module from inside a jar file, or a physical file path.

I've got to a point where the game works from the development environment, and also after packaging and extracting, but I don't have the capacity to test this on Android, so in the meantime you can check it out in my repo https://github.com/Avalancs/DestinationSol/tree/DesktopFix

**Comment by Cervator** [comment]
Fixed now with #415!
